packages/server/graphql/httpGraphQLHandler.ts

```diff
@@ -64,21 +65,28 @@ const httpGraphQLBodyHandler = async (
   })
 }
 
+const contentTypeBodyParserMap = {
+  'application/json': parseBody,
+  'multipart/form-data': parseFormBody
+}
+
 const httpGraphQLHandler = uWSAsyncHandler(async (res: HttpResponse, req: HttpRequest) => {
-  const contentType = req.getHeader('content-type')
   const connectionId = req.getHeader('x-correlation-id')
   const authToken = getReqAuth(req)
   const ip = uwsGetIP(res, req)
-  if (contentType.startsWith('application/json')) {
-    const body = await parseBody(res)
-    if (!body) {
-      res.writeStatus('422').end()
-      return
-    }
-    await httpGraphQLBodyHandler(res, body, authToken, connectionId, ip)
-  } else {
+  const contentTypeHeader = req.getHeader('content-type')
+  const ct = Object.keys(contentTypeBodyParserMap).find((key) => contentTypeHeader.startsWith(key))
+  if (!ct) {
     res.writeStatus('415').end()
+    return
+  }
+  const parseFn = contentTypeBodyParserMap[ct]
+  const body = await parseFn({res, req})
```

Comment:
call signature of `parseBody` changed so that we can have a unified call to whichever fn here.

Reply:
that is very very good to know. so any data from the req we want to hold onto for async things we do via closures right? i'll hold off on fixing this until we get the uwebsockets upgraded

Reply:
on second thought, i'm probably in agreement with:

> it might be more expressive to use a different function call/signature for each

just for the fact of getting deterministic autocompletions for each function.